openedx/openedx-platform
# feat: update preview url to direct to mfe

## Description

This PR changes unit preview URL. Currently, unit previews render in the legacy learners page. As the learning MFE is developed and refiend, the UI between the legacy page and the MFE begins to diverge. This make it difficult for author to accurately determine what their content will look like for learners before it is published. With the PRs change, unit previews will no longer redirect to the legacy page, but the learning MFE. Additionally, this PR updates the content of the unit iframe depending on if it is in a preview or learner view. This change impact Authors. Some follow-up work may be needed to delete config for `preview.host`

## Supporting information

JIRA Ticket: [AU-1067 🔒](https://2u-internal.atlassian.net/browse/AU-1067)

> The Preview environment has drifted a bit away from the live view. The particular thing I spotted was that all the unit content is in an iframe, and preview doesn’t. There are some visual differences too, like the size of navigation elements, the placement of the bottom-of-page Previous/Next buttons, etc. 
>
> Things still look fairly accurate, but if you’re testing javascript, it can be a big difference.

## Testing instructions

1. Open a course in Studio
2. Publish an existing unit
3. Make a change to the unit so that it is showing a draft version in Studio
4. Click "View live"
5. Confirm that the published version of unit is visible
6. Confirm that preview is not in the URL
7. Go back to the Studio page
8. Click "Preview"
9. Confirm that the draft version of the unit is visible
10. Confirm that "preview" is at the beginning of the URL

## Deadline

None

## Other information

This PR is dependent on `frontend-app-learning` PR [#1501](https://github.com/openedx/frontend-app-learning/pull/1501)
